src/cartridges/int_adyen_SFRA/cartridge/adyen/scripts/expressPayments/paypal/makeExpressPaymentDetailsCall.js

```diff
@@ -0,0 +1,69 @@
+const URLUtils = require('dw/web/URLUtils');
+const OrderMgr = require('dw/order/OrderMgr');
+const Transaction = require('dw/system/Transaction');
+const BasketMgr = require('dw/order/BasketMgr');
+const adyenCheckout = require('*/cartridge/adyen/scripts/payments/adyenCheckout');
+const AdyenLogs = require('*/cartridge/adyen/logs/adyenCustomLogs');
+const COHelpers = require('*/cartridge/scripts/checkout/checkoutHelpers');
+const AdyenHelper = require('*/cartridge/adyen/utils/adyenHelper');
+const paypalHelper = require('*/cartridge/adyen/utils/paypalHelper');
+const constants = require('*/cartridge/adyen/config/constants');
+
+function setPaymentInstrumentFields(paymentInstrument, response) {
+  paymentInstrument.custom.adyenPaymentMethod =
+    AdyenHelper.getAdyenComponentType(response.paymentMethod.type);
+  paymentInstrument.custom[`${constants.OMS_NAMESPACE}__Adyen_Payment_Method`] =
+    AdyenHelper.getAdyenComponentType(response.paymentMethod.type);
+  paymentInstrument.custom.Adyen_Payment_Method_Variant =
+    response.paymentMethod.type.toLowerCase();
+  paymentInstrument.custom[
+    `${constants.OMS_NAMESPACE}__Adyen_Payment_Method_Variant`
+  ] = response.paymentMethod.type.toLowerCase();
+}
+
+/*
+ * Makes a payment details call to Adyen to confirm the current status of a payment.
+   It is currently used only for PayPal Express Flow
+ */
+function makeExpressPaymentDetailsCall(req, res, next) {
+  try {
+    const request = JSON.parse(req.body);
+    const currentBasket = BasketMgr.getCurrentBasket();
+
+    const response = adyenCheckout.doPaymentsDetailsCall(request.data);
+
+    paypalHelper.setBillingAndShippingAddress(currentBasket);
+
+    // Setting the session variable to null after assigning the shopper data to basket level
+    session.privacy.shopperDetails = null;
```

Comment:
Is this okay in the sense it will not affect other parts of the application unintentionally?

Reply:
This is used only for PayPal express and not for other payment method. Once we assign the data in basket level, we need to clean